sky/skylet/providers/gcp/config.py

```diff
@@ -994,19 +992,26 @@ def _list_vpcnets(config, compute, filter=None):
 
 
 def _list_subnets(
-    config, compute, filter=None
+    config, compute, network=None
 ) -> List["google.cloud.compute_v1.types.compute.Subnetwork"]:
     response = (
         compute.subnetworks()
         .list(
             project=config["provider"]["project_id"],
             region=config["provider"]["region"],
-            filter=filter,
         )
         .execute()
     )
 
-    return response["items"] if "items" in response else []
+    items = response["items"] if "items" in response else []
+    if network is None:
```

Comment:
Just curious, is it possible to do the network filtering with the filter field? If not, maybe we can have a comment here?

Reply:
Added a comment:
```
    # Filter by network (VPC) name.
    #
    # Note we do not directly use the filter (network=<...>) arg of the list()
    # call above, because it'd involve constructing a long URL of the following
    # format and passing it as the filter value:
    # 'https://www.googleapis.com/compute/v1/projects/<project_id>/global/networks/<network_name>'
```